File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        L, R = 0, len(s) - 1

        while L <= R:
            if not s[L].isalnum(): # if it is not both alphabetic nor numeric use the method isalnum() or use isalpha() and isnumeric() or isdigit() together
                L += 1
                continue
            if not s[R].isalnum():
                R -= 1
                continue
            if (s[L].lower()) != (s[R].lower()):
                return False
            L += 1
            R -= 1
        
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int L = 0, R = s.length() - 1;

        while (L < R)
        {
            if (L < R && !(isalpha(s[L]) || isdigit(s[L]))) // or isalnum(s[L])
            {
                L++;
                continue;
            }
            if (L < R && !(isalpha(s[R]) || isdigit(s[R])))
            {
                R--;
                continue;

            }
            if (tolower(s[L]) != tolower(s[R]))
               return false;
            L++, R--;
        }
        return true;
    }
};